### PR TITLE
KEYCLOAK-12165 Fix UserSessionProviderTest to work correctly

### PR DESCRIPTION
**Pull request by y-tabata**
Changed `createSessions` method to commit correctly, and then fixed failed tests (`testRemoveUserSessionsByUser` and `testRemoveUserSessionsByRealm`).

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java
Comment (tnorimat) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java]: Please note that there are two types of method in InfinispanUserSessionProvider.
* check entities on transactions at first, if not found, next check on the cache (ex. getUserSession).
* check entities only on the cache (ex. getUserSessions)

To clarify whether each test method of this class try to check entities on only the cache or not, I think it needs to consider the point above.

Comment (tnorimat) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java]: It seems that on each node keycloak keeps several changes against entities that are managed on the cache and commit them in the bulk. It improves performance on cache manipulation.
Those changes are managed by changelog transaction (InfinispanChangelogBasedTransaction).
KeycloakTransactionManager.commit() commits those changes.

Comment (y-tabata) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java]: @tnorimat thanks! I will change only the uncorrect tests (`testRemoveUserSessionsByUser` and `testRemoveUserSessionsByRealm`) and not change `createSessions`.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java
Comment (y-tabata) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java]: Changed.

**Comment by tnorimat** [comment]
@y-tabata I've not yet understood keycloak's infinispan cache management completely, but your changes seem good.

**Review by stianst — DISMISSED** [review]

**Comment by stianst** [comment]
LGTM, but needs a rebase.

@mposolda do you want to take a quick look?

**Review by mposolda — DISMISSED** [review]

**Comment by mposolda** [comment]
@stianst I've also approved. After rebase, this can be merged IMO.

**Comment by y-tabata** [comment]
@mposolda @stianst thanks! I rebased.

**Comment by y-tabata** [comment]
@stianst @mposolda thanks! I rebased.

**Review by stianst — APPROVED** [review]

**Comment by stianst** [comment]
LGTM @mposolda can you merge if you are happy with it?

**Review by mposolda — APPROVED** [review]

**Pull request merged**
